.github/workflows/test.yml

```diff
@@ -0,0 +1,106 @@
+name: "Test aws-robomaker-github-actions"
+
+on:
+  push:
+  pull_request:
+    branches:
+      - master
+      - 'releases/*'
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  test_robomaker-sample-app-ci:
+    runs-on: ubuntu-latest
+    
+    strategy:      
+      matrix:
+        distro: ['kinetic', 'melodic']
+        gazebo: [7, 9]
+        include:
+        - distro: kinetic
+          gazebo: 7
+          ubuntu_distro: xenial
+        - distro: kinetic
+          gazebo: 9
+          ubuntu_distro: xenial
+        - distro: melodic
+          gazebo: 9
+          ubuntu_distro: bionic
+        exclude:
+        - distro: 'melodic'
+          gazebo: 7      
+      
+    container: 
+      image: rostooling/setup-ros-docker:ubuntu-${{ matrix.ubuntu_distro }}-ros-${{ matrix.distro }}-ros-base-latest
+      
+    # NOTES: We run 2 tests for robot_ws and simulation_ws
+    # The test steps are duplicated because github workflow does not support reusable step (with parameter yet)
+    # We also can use matrix to run tests with different parameters but it is overkill for this
+    steps:    
+    - name: Checkout code
+      uses: actions/checkout@v1
+    - name: Setup permissions
+      run: |
+        # Due to user permisson issue, calling chown is necessary for now
+        # Related issue: https://github.com/actions/checkout/issues/47
+        # TODO(ros-tooling/setup-ros-docker#7): 
+        sudo chown -R rosbuild:rosbuild "$HOME" .      
+    - name: Setup nodeJS
+      uses: actions/setup-node@v1
+      with:
+        node-version: '12.x'
+    - name: Install nodeJS packages
+      run: npm install
+      working-directory: ./robomaker-sample-app-ci
+    - name: Build test script
+      run: npm run build
+      working-directory: ./robomaker-sample-app-ci
+      
+    # Trigger to run robomaker-sample-app-ci action script
+    - name: Build and bundle robot_ws      
+      run: node ./robomaker-sample-app-ci/dist/index.js
+      env:
+        INPUT_ROS-DISTRO: ${{ matrix.distro }}
+        INPUT_GAZEBO-VERSION: ${{ matrix.gazebo }}
+        INPUT_WORKSPACE-DIR: ./.github/workflows/data/robot_ws
```

Comment:
Should we be fetching the latest source files for build and bundle from the SA github repos? I think it's a bit tricky as changes in SA repos should not break the CI tests for these actions but as we last discussed, any breaking changes would <em>never be pushed</em> to the head of SA repos. With current setup, there is a chance that these tests might get redundant if the SA repo changes.

Reply:
I tested with SA..but here I am trying to minimize the number of dependencies. Integration test itself already has enough complexity (run docker image, install ros...) the more dependencies we have the higher chance it will break. So I decided to include very simple example app embedded in the test data instead. The goal of the tests to verify that the action itself working as expected, not to tight it to any SA.

Reply:
Yes I agree with decoupling these tests and actions from SA's.. robomaker-sample-app-ci action in itself is however not decoupled entirely (for eg: [getSampleVersion()](https://github.com/aws-robotics/aws-robomaker-github-actions/blob/03720c02e1af2052ba7a59140c0b00b17d302ac2/robomaker-sample-app-ci/src/aws-robomaker-sample-application-ci.ts#L52))..so just want to raise the question -- in case a breaking change is made in a coupled function within the actions here, do we want the CI to break here or in the Sample App?

Reply:
To me, SA is just general term we are using for multiple sample app(s) that have this structure: robot_ws, simulation_ws... (aws-robomaker-sample-application-helloworld, aws-robomaker-sample-application-cloudwatch). The embedded SA follows that structure, so getSampleVersion is not a problem. If there is breaking change in this repos, the CI of this repos should catch it. This is also good question because it raises another issue that was addressed in another discussion (stability and security). I think if we really want to address the stability, we should use versioning tag (or commit SHA) as  recommended by Github (making change in action, test it, tag it as major/minor version... and make change to SA as needed if SA's CI wants to use newer versions)

Reply:
As far as my knowledge, we do have versioning tags for actions and the corresponding version tags are being used in the GH actions in Cloudwatch and Helloworld SAs.

Reply:
With my [comment above](https://github.com/aws-robotics/aws-robomaker-github-actions/pull/9/files#r488856420), I would prefer us using our sample apps for this.

@wangvnn if you're concerned with sample apps constantly changing such that there's too many moving parts to this test, you can fetch a specific tag or commit of the sample apps instead.

Reply:
The issue is not about which version of SA...but the complexity of SA. SA can do more things than we need (e.g pull in turtle bot source from github) if that fails we fails our CI here and that is unnecessary. I built the embedded test app from scratch not pulling from any source.

Reply:
Pulling TurtleBot sources (executing `vcs import`) is the responsibility of this GitHub action. If you remove that complexity, I think you won't be testing this GitHub action sufficiently.

Reply:
Yep, I think I missed that vcs import. I was even thinking about a simpler .rosinstall to pull in something simpler than turtlebot. 
It looks like team prefer pulling the SA. I will make the change. Which one do you prefer for this test to pull in (hello world or cloud watch or both) ?

Reply:
In terms of code coverage, I think either sample app will exercise the same amount of code in this GitHub action. Let's go with just the Hello World sample app, since it should build and bundle slightly quicker.

Reply:
new update: using hello world app for testing now.